### PR TITLE
fix: Interface field call is redundant

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <farmer.hutao@outlook.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

fix: Interface field call is redundant, it should has consistent style with q.Add() below

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by cndoit18 — APPROVED** [review]
LGTM

**Comment by daniel-hutao** [comment]
/assign @lavalamp
please take a look if you have time, thx

File: staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func (q *delayingType) waitingLoop() {
 	waitingEntryByData := map[t]*waitFor{}
 
 	for {
-		if q.Interface.ShuttingDown() {
+		if q.ShuttingDown() {
```
Comment (RyanAoh) [inline staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go:202]: As `q.Interface` is an embedded field, I think both the two ways to call  `ShuttingDown` are ok via https://golang.google.cn/doc/effective_go#embedding. So I wonder if it's necessary to make this change?  Correct me if I misunderstand.

Comment (daniel-hutao) [reply staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go:202]: There is no difference between the two coding style at functional, but I think it is more harmonious to keep the style consistent; Other methods called for q do not have an Interface field:

- `q.Add(entry.data)` at line 216 245 255
- `q.ShuttingDown()` at line 162
- `q.Add(item)` at line 170

Comment (RyanAoh) [reply staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go:202]: If so, line 153 should be changed, too.

Comment (daniel-hutao) [reply staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go:202]: I'll do it right now

Comment (daniel-hutao) [reply staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go:202]: > If so, line 153 should be changed, too.

153 can't be changed for the same name method: `func (q *delayingType) ShutDown()`

Comment (RyanAoh) [reply staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go:202]: Oh，yeah. It's my negligence.

Comment (RyanAoh) [reply staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go:202]: Uh，yeah. It's my negligence.

**Comment by lavalamp** [comment]
Personally, I don't think this is the change this file needs, on reading the code:
1) I'd delete the call to `q.ShuttingDown` on line 162 (the whole if block), it's racy
2) I'd make the `waitingLoop` function defer a call to `ShutDown()`

There could be a bug that causes AddAfter to never return if the Interface's ShutDown method is called directly, and waitingForAddCh is full... but the constructor means that callers shouldn't be able to separately shut down the inner queue. So I'm not sure it's worth fixing, although it's bugging me a lot now that you've brought it to my attention. :)

Stylistically, I personally think `q.Interface.ShuttingDown()` makes it clearer what is going on.

**Comment by dims** [comment]
@daniel-hutao will you be coming back to this PR? please see the last set of feedback from reviewers

**Comment by daniel-hutao** [comment]
> @daniel-hutao will you be coming back to this PR? please see the last set of feedback from reviewers

@dims 
Yeah, I will keep watching. I didn't read the above description carefully before, because reading English is not too smooth for me, so I only noticed "I personally think q.Interface.ShuttingDown() makes it clearer what is going on." before. I just took a closer look:

1. I agree with keeping `q.Interface.ShuttingDown()` for it is more clearer.
2. Regarding the refactor(or bugfix?) mentioned above, I thought @lavalamp  was going to handle it. Can I participate in this part?

**Comment by dims** [comment]
@daniel-hutao yes, please close this and open a PR for refactoring that Daniel mentions

**Comment by daniel-hutao** [comment]
> @daniel-hutao yes, please close this and open a PR for refactoring that Daniel mentions

Ok, I'll close this pr after the new PR is opened

**Pull request closed without merging**